### PR TITLE
bottle: add tests for `merge_bottle_spec` helper method

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

This PR adds more checks in `merge_bottle_spec` for default values and adds tests to make sure that the checks work

Related: https://github.com/Homebrew/brew/pull/10102, https://github.com/Homebrew/brew/pull/10130

### CODE REVIEW

File: Library/Homebrew/dev-cmd/bottle.rb [hunk 1]
```diff
@@ -546,13 +546,23 @@ def merge_bottle_spec(old_bottle_spec, new_bottle_hash)
...
       old_value = old_bottle_spec.send(key)
       next if key == :rebuild && old_value.zero?
-      next if key == :prefix && old_value == Homebrew::DEFAULT_PREFIX
+      next if key == :root_url && [
+        HOMEBREW_BOTTLE_DEFAULT_MACOS_DOMAIN,
+        HOMEBREW_BOTTLE_DEFAULT_LINUX_DOMAIN,
+      ].any? { |domain| old_value.start_with? domain }
+      next if key == :prefix && [
+        Homebrew::DEFAULT_PREFIX,
+        HOMEBREW_DEFAULT_PREFIX,
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bottle.rb:560]:
```suggestion
        HOMEBREW_DEFAULT_PREFIX,
```

Don't need both of these.

**Review by MikeMcQuaid — APPROVED** [review]
👏🏻

File: Library/Homebrew/dev-cmd/bottle.rb (hunk 1 above)
Comment (SeekingMeaning) [reply Library/Homebrew/dev-cmd/bottle.rb:560]: It's required for tests as it is redefined here: https://github.com/Homebrew/brew/blob/2c77a54/Library/Homebrew/test/support/lib/config.rb#L53-L54

Should I instead redefine `HOMEBREW_DEFAULT_PREFIX` as well?

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/bottle.rb:560]: Hmm, I wonder if we can get it down to a single variable here then by replacing all occurrences of one with the other?

Comment (iMichka) [reply Library/Homebrew/dev-cmd/bottle.rb:560]: Should we investigate more here or can we merge this PR? I would like to open my PR for the new cellar syntax, but as long as all of this code is a moving target it's not possible for me.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/bottle.rb:560]: I'd like to get it sorted before this PR is merged ideally, sorry.

**Comment by SeekingMeaning** [comment]
The current logic of checking for default values feels slightly wrong. I’m going to open a new PR that re-implements the old logic of checking what methods are called in the `bottle` block

**Pull request closed without merging**